gaeguli/streamadaptor.c

```diff
@@ -0,0 +1,179 @@
+
+  if (klass->on_stats) {
+    gaeguli_stream_adaptor_start_timer (self);
+  }
+}
+
+static void
+gaeguli_stream_adaptor_set_stats_interval (GaeguliStreamAdaptor * self,
+    guint seconds)
+{
+  GaeguliStreamAdaptorPrivate *priv =
+      gaeguli_stream_adaptor_get_instance_private (self);
+
+  priv->stats_interval = seconds;
+
+  if (priv->stats_timeout_id != 0) {
+    g_clear_handle_id (&priv->stats_timeout_id, g_source_remove);
+    gaeguli_stream_adaptor_start_timer (self);
+  }
+}
+
+static void
+gaeguli_stream_adaptor_init (GaeguliStreamAdaptor * self)
+{
+}
+
+static void
+gaeguli_stream_adaptor_set_property (GObject * object, guint property_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GaeguliStreamAdaptor *self = GAEGULI_STREAM_ADAPTOR (object);
+
+  switch (property_id) {
+    case PROP_SRTSINK:
+      gaeguli_stream_adaptor_set_srtsink (self, g_value_get_object (value));
+      break;
+    case PROP_STATS_INTERVAL:
+      gaeguli_stream_adaptor_set_stats_interval (self,
+          g_value_get_uint (value));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+  }
+}
+
+static void
+gaeguli_stream_adaptor_get_property (GObject * object, guint property_id,
+    GValue * value, GParamSpec * pspec)
+{
+  GaeguliStreamAdaptor *self = GAEGULI_STREAM_ADAPTOR (object);
+  GaeguliStreamAdaptorPrivate *priv =
+      gaeguli_stream_adaptor_get_instance_private (self);
+
+  switch (property_id) {
+    case PROP_STATS_INTERVAL:
+      g_value_set_uint (value, priv->stats_interval);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+  }
+}
+
+static void
+gaeguli_stream_adaptor_dispose (GObject * object)
+{
+  GaeguliStreamAdaptor *self = GAEGULI_STREAM_ADAPTOR (object);
+  GaeguliStreamAdaptorPrivate *priv =
+      gaeguli_stream_adaptor_get_instance_private (self);
+
+  gst_clear_object (&priv->srtsink);
+  g_clear_handle_id (&priv->stats_timeout_id, g_source_remove);
+
+  G_OBJECT_CLASS (gaeguli_stream_adaptor_parent_class)->dispose (object);
+}
+
+static void
+gaeguli_stream_adaptor_class_init (GaeguliStreamAdaptorClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+
+  gobject_class->set_property = gaeguli_stream_adaptor_set_property;
+  gobject_class->get_property = gaeguli_stream_adaptor_get_property;
+  gobject_class->dispose = gaeguli_stream_adaptor_dispose;
+
+  g_object_class_install_property (gobject_class, PROP_SRTSINK,
+      g_param_spec_object ("srtsink", "SRT sink",
+          "SRT sink", GST_TYPE_ELEMENT,
+          G_PARAM_WRITABLE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_STATS_INTERVAL,
+      g_param_spec_uint ("stats-interval", "Statistics collection interval",
+          "Statistics collection interval in seconds", 1, G_MAXUINT, 5,
+          G_PARAM_READWRITE | G_PARAM_CONSTRUCT | G_PARAM_STATIC_STRINGS));
```

Comment:
If it means sampling rate, one second is too sparse for our use case. 
We need to support the 10-millisecond grade sampling rate.

Reply:
Alright, I changed this to have the statistics collection trigger every 10ms.